sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/physical/partitioning.scala

```diff
@@ -118,10 +115,12 @@ case class HashClusteredDistribution(
 
 /**
  * Represents data where tuples have been ordered according to the `ordering`
- * [[Expression Expressions]].  This is a strictly stronger guarantee than
- * [[ClusteredDistribution]] as an ordering will ensure that tuples that share the
- * same value for the ordering expressions are contiguous and will never be split across
- * partitions.
+ * [[Expression Expressions]]. Its requirement is defined as the following:
+ *   - Given any 2 adjacent partitions, all the rows of the second partition must be larger than or
+ *     equal to any row in the first partition, according to the `ordering` expressions.
```

Comment:
Why here we need this equality? Can we just have all the rows in the second partition must be larger than any row in the first partition?

Reply:
Global sort (actually the `RangePartitioner`) currently guarantees that all rows in partition `p + 1` are larger than the rows in partition `p`. I don't think we should relax this, besides collect limit there aren't any use cases I can think of that could work with this relaxed requirement.

Reply:
Let us keep the semantics unchanged at this moment. If needed, in the future, we can either introduce a new distribution type or change the existing types.

Reply:
@hvanhovell We need this relaxed requirement, otherwise we have to remove the optimization [here](https://github.com/apache/spark/pull/23249/files#diff-14e5db1a500000efdaf9c760aeb5ced4R273)

Reply:
I did not change the semantic, I just correct the comment to represent what the current semantic is.

Reply:
Yes, @cloud-fan is right about the "or equal to" part is necessary for RangePartitioning(a, b, c) satisfying OrderedDistribution(a, b).